pkg/synthetictests/kubelet.go

```diff
@@ -363,3 +364,70 @@ func testSystemDTimeout(events monitorapi.Intervals) []*ginkgo.JUnitTestCase {
 	// write a passing test to trigger detection of this issue as a flake. Doing this first to try to see how frequent the issue actually is
 	return []*ginkgo.JUnitTestCase{failure, success}
 }
+
+var errImagePullTimeoutRE = regexp.MustCompile("ErrImagePull.*read: connection timed out")
+var errImagePullGenericRE = regexp.MustCompile("ErrImagePull")
+
+func testErrImagePullConnTimeoutOpenShiftNamespaces(events monitorapi.Intervals) []*ginkgo.JUnitTestCase {
+	const testName = "[sig-node] should not encounter ErrImagePull read connection timeout in openshift namespace pods"
+	return buildTestsFailIfRegexMatch(testName, errImagePullTimeoutRE, nil, true, events)
+}
+
+func testErrImagePullConnTimeout(events monitorapi.Intervals) []*ginkgo.JUnitTestCase {
+	const testName = "[sig-node] should not encounter ErrImagePull read connection timeout in non-openshift namespace pods"
+	return buildTestsFailIfRegexMatch(testName, errImagePullTimeoutRE, nil, false, events)
```

Comment:
`nil, false` reads really badly. if you make a struct to hold your three configuration settings you could consistently name them.  If you made a const in here, you coudl refer to it by name in-line

Reply:
I tried the struct but not a fan of how that's coming out, everything then starts looking optional when in reality, only the dontMatchRegex is optional, and then the function starts having to do a lot of nil/empty checking and error handling because it's not as clear how to invoke it. More than one optional param and I can see the value in the struct.

I added an enum for the in openshift- or not bool instead though. Hopefully this is clear enough what is being done.